### PR TITLE
[API] Mapping/serialization cleanup

### PR DESCRIPTION
**Pull request by NoResponseMate**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.13 |
| Bug fix?        | no                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no  |
| Related tickets | -                      |
| License         | MIT                                                          |

### CODE REVIEW

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/TaxonTranslation.xml [hunk 1]
```diff
@@ -7,18 +7,7 @@
...
-                <attribute name="path">/admin/taxon-translations</attribute>
-                <attribute name="normalization_context">
-                    <attribute name="groups">
-                        <attribute>admin:taxon:read</attribute>
-                        <attribute>admin:taxon_translation:read</attribute>
-                    </attribute>
-                </attribute>
-            </collectionOperation>
-        </collectionOperations>
+        <collectionOperations />
```
Comment (coldic3) [inline src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/TaxonTranslation.xml:10]: What about normalization groups in this case?

**Review by coldic3 — COMMENTED** [review]
In the future, we can think about using `ApiPlatform\Serializer\SerializerContextBuilderInterface` to add normalization groups automatically to the context.

File: src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/TaxonTranslation.xml (hunk 1 above)
Comment (NoResponseMate) [reply src/Sylius/Bundle/ApiBundle/Resources/config/api_resources/TaxonTranslation.xml:10]: The endpoint itself went away. This was the only place where translations were treated differently and had an index, despite being included in the Taxon serialization.

File: tests/Api/Admin/CurrenciesTest.php [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+use Sylius\Tests\Api\JsonApiTestCase;
+use Sylius\Tests\Api\Utils\AdminUserLoginTrait;
+use Symfony\Component\HttpFoundation\Response;
+
+final class CurrenciesTest extends JsonApiTestCase
+{
+    use AdminUserLoginTrait;
+
+    /** @test */
+    public function it_gets_a_country(): void
```
Comment (jakubtobiasz) [inline tests/Api/Admin/CurrenciesTest.php]:
```suggestion
    public function it_gets_a_currency(): void
```

File: tests/Api/Admin/ProvincesTest.php [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+use Sylius\Tests\Api\JsonApiTestCase;
+use Sylius\Tests\Api\Utils\AdminUserLoginTrait;
+use Symfony\Component\HttpFoundation\Response;
+
+final class ProvincesTest extends JsonApiTestCase
+{
+    use AdminUserLoginTrait;
+
+    /** @test */
+    public function it_gets_a_country(): void
```
Comment (jakubtobiasz) [inline tests/Api/Admin/ProvincesTest.php]:
```suggestion
    public function it_gets_a_province(): void
```

**Review by jakubtobiasz — APPROVED** [review]
I've just found only two typos c:.

File: src/Sylius/Bundle/ApiBundle/Resources/config/serialization/Country.xml [hunk 4]
```diff
@@ -16,9 +16,6 @@
             xsi:schemaLocation="http://symfony.com/schema/dic/serializer-mapping https://symfony.com/schema/dic/serializer-mapping/serializer-mapping-1.0.xsd"
 >
     <class name="Sylius\Component\Addressing\Model\Country">
-        <attribute name="id">
-            <group>admin:country:read</group>
-        </attribute>
```
Comment (GSadee) [inline src/Sylius/Bundle/ApiBundle/Resources/config/serialization/Country.xml]: Why are you removing the `id` from this one and some other resources?

File: tests/Api/Admin/ProductReviewsTest.php [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+use Sylius\Tests\Api\JsonApiTestCase;
+use Sylius\Tests\Api\Utils\AdminUserLoginTrait;
+use Symfony\Component\HttpFoundation\Response;
+
+final class ProductReviewsTest extends JsonApiTestCase
+{
+    use AdminUserLoginTrait;
+
+    /** @test */
+    public function it_gets_product_review(): void
```
Comment (GSadee) [inline tests/Api/Admin/ProductReviewsTest.php]:
```suggestion
    public function it_gets_a_product_review(): void
```

File: tests/Api/Admin/PromotionTest.php [hunk 6]
```diff
@@ -13,6 +13,7 @@
```
Comment (GSadee) [inline tests/Api/Admin/PromotionTest.php]: You should probably change the name of this file to `PromotionsTest` to make it consistent with others

File: tests/Api/Admin/ShippingMethodsTest.php [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+
+        $this->assertResponse(
+            $this->client->getResponse(),
+            'admin/shipping_method/get_shipping_methods_response',
+            Response::HTTP_OK,
+        );
+    }
+
+    /** @test */
+    public function it_archives_a_shipping_methods(): void
```
Comment (GSadee) [inline tests/Api/Admin/ShippingMethodsTest.php]:
```suggestion
    public function it_archives_a_shipping_method(): void
```

File: tests/Api/Admin/ShippingMethodsTest.php [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+
+        $this->assertResponse(
+            $this->client->getResponse(),
+            'admin/shipping_method/archive_shipping_method_response',
+            Response::HTTP_OK,
+        );
+    }
+
+    /** @test */
+    public function it_restores_a_shipping_methods(): void
```
Comment (GSadee) [inline tests/Api/Admin/ShippingMethodsTest.php]:
```suggestion
    public function it_restores_a_shipping_method(): void
```

File: src/Sylius/Bundle/ApiBundle/Resources/config/serialization/Country.xml (hunk 4 above)
Comment (NoResponseMate) [reply src/Sylius/Bundle/ApiBundle/Resources/config/serialization/Country.xml]: IMO, these shouldn't be included in the API context when another field is used as the identifier, seems like dead weight. It's not really a strong opinion so I've reverted it. We can always discuss it further, but either way, a change like that should go in a separate PR for transparency's sake.

**Review by GSadee — APPROVED** [review]

**Comment by GSadee** [comment]
Thank you, Jan! :1st_place_medal:

**Pull request merged**